quinn-proto/src/crypto/rustls.rs

```diff
@@ -263,7 +266,11 @@ impl crypto::ClientConfig for rustls::ClientConfig {
                 )
                 .unwrap(),
             ),
-        }))
+        };
+        #[cfg(not(feature = "dangerous_packet_null_cipher"))]
```

Comment:
The way this is implemented makes this a global setting. Either all quinn implementations inside an executable would use real crypto, or they would use no crypto. It's not a per-endpoint or per-connection choice anymore.

This is a bit inflexible.

But what is even more problematic is that it opens the door for supply-chain attacks. You can have any other crate define a dependency on quinn with the dangerous feature enabled, and as a result our code which directly uses quinn and the intention of being safe now was crypto enabled.

I think what would be safer is the rustls approach of requiring the feature flag to enable a new API which actually disables the security. So there should be an EndpointConfig/TransportConfig/etc that decides whether to return `NullCipherSession` or not.